web/auth/auth_test.go

```diff
@@ -2,425 +2,289 @@ package auth_test
 
 import (
 	"net/http"
-	"net/http/httptest"
-	"reflect"
 	"testing"
 
-	"github.com/gorilla/sessions"
-	"github.com/labstack/echo-contrib/session"
-	"github.com/labstack/echo/v4"
 	"github.com/markbates/goth"
-	"github.com/markbates/goth/gothic"
+	"github.com/quickfeed/quickfeed/database"
 	"github.com/quickfeed/quickfeed/internal/qtest"
-	"github.com/quickfeed/quickfeed/qf"
 	"github.com/quickfeed/quickfeed/qlog"
 	"github.com/quickfeed/quickfeed/web/auth"
+	"github.com/steinfletcher/apitest"
+	"gotest.tools/assert"
 )
 
 const (
-	loginRedirect  = "/login"
-	logoutRedirect = "/logout"
-
-	authURL   = "/auth?provider=fake&redirect=" + loginRedirect
-	logoutURL = "/logout?provider=fake&redirect=" + logoutRedirect
-
-	fakeSessionKey  = "fake"
-	fakeSessionName = fakeSessionKey + gothic.SessionName
+	testSecret = "top-secret"
 )
 
 func init() {
 	goth.UseProviders(&auth.FakeProvider{
-		Callback: "/auth/fake/callback",
+		Callback: "/auth/callback/fake",
 	})
 }
 
-func TestOAuth2Logout(t *testing.T) {
-	r := httptest.NewRequest(http.MethodGet, logoutURL, nil)
-	w := httptest.NewRecorder()
-
-	store := newStore()
-	gothic.Store = store
-
-	e := echo.New()
-	c := e.NewContext(r, w)
-
-	fakeSession := auth.FakeSession{}
-	s, _ := store.Get(r, fakeSessionName)
-	s.Values[fakeSessionKey] = fakeSession.Marshal()
-	if err := s.Save(r, w); err != nil {
-		t.Error(err)
-	}
-
-	if err := store.login(c); err != nil {
-		t.Error(err)
-	}
-
-	ns := len(store.store[r].Values)
-	// Want gothic session and user session.
-	if ns != 2 {
-		t.Errorf("have %d sessions want %d", ns, 2)
-	}
-
-	authHandler := auth.OAuth2Logout(qlog.Logger(t))
-	withSession := session.Middleware(store)(authHandler)
-
-	if err := withSession(c); err != nil {
-		t.Error(err)
-	}
-
-	ns = len(store.store[r].Values)
-	// Sessions should be cleared.
-	if ns != 0 {
-		t.Errorf("have %d sessions want %d", ns, 0)
-	}
+func TestOAuth2Login(t *testing.T) {
+	logger := qlog.Logger(t)
+	authConfig := auth.NewGitHubConfig("", "TEST", "TEST")
+	// Incorrect request method.
+	apitest.New().Debug().
```

Comment:
I like this new testing framework. :-) 

However, is there a way to turn off debug mode? E.g., so that we can run `go test -v` to enable debug mode (or more verbose output)...?